napari_aicssegmentation/controller/workflow_steps_controller.py

```diff
@@ -31,9 +33,24 @@ def close_workflow(self):
         self.model.reset()
         self.router.workflow_selection()
 
-    def run_all(self):
-        result = self.model.active_workflow.execute_all()
-        self.viewer.add_image(result, name=f"Result for workflow {self.model.active_workflow.name}")
+    def run_all(self, parameter_inputs: List[Dict[str, List]]):
+        """ 
+        parameter_inputs is a list of dictionaries, one dictionary (or None if no parameters) 
+        for each step. Each dictionary has the same shape as a WorkflowStep.parameter_defaults
+        dictionary, but with the parameter values obtained from the UI instead of default values.
+        """
+
+        workflow = self.model.active_workflow
+
+        # This is equivalent to running workflow.execute_all(), but with non-default params
+        workflow.reset()
```

Comment:
wait what does this do?

Reply:
It resets the state of the workflow so that it starts at Step 1 again. It's needed if you want to re-run the workflow, otherwise it'll just stay in "done" status.

Reply:
Not absolutely positive but it was part of `workflow.execute_all()` in the Segmenter code (which I'm replacing here) so I copied it over!

Reply:
Yes it seems necessary :) https://github.com/AllenCell/aics-segmentation/blob/feature/workflow-engine/aicssegmentation/workflow/workflow.py#L32